src/integrations/fhir/ProxiedFhirClientWrapper.ts

```diff
@@ -126,6 +134,54 @@ export default class ProxiedFhirClientWrapper implements FhirApi {
         return relatedPersons
     }
 
+    public blobToBase64 = (file: Blob): Promise<string> =>
+        new Promise((resolve, reject) => {
+            const reader = new FileReader();
+            reader.onload = () => {
+                const result = reader.result as string;
+                resolve(result.split(",")[1]);
+            };
+
+            reader.readAsDataURL(file);
+            reader.onerror = reject;
+        });
+
+    public async createDocument(patientEhrId: string, providerEhrId: string, hospitalEhrId: string, pdf: Blob): Promise<IDocumentReference> {
+        const pdfAsBase64 = await this.blobToBase64(pdf);
+        const docunmentReference = createAndValidateDocumentReferencePayload(
+            patientEhrId,
+            providerEhrId,
+            hospitalEhrId,
+            DocumentReferenceStatusKind._current,
+            [
+                {
+                    "attachment": {
+                        "contentType": "application/pdf",
+                        "data": pdfAsBase64,
+                        "title": "title",
+                    }
+                }
+            ]
+        )
+
+        return this.client.request<IDocumentReference>(
+            {
+                url: "DocumentReference",
+                method: "POST",
+                body: JSON.stringify(docunmentReference),
+                headers: {
+                    "Content-Type": "application/json",
+                },
+            }
+        ).then((data: IDocumentReference) => {
```

Comment:
Er ikkje denne logginga berre støy når ein har fått det til å fungere? Feil som blir kasta viser vel uansett i konsoll, og data som blir returnert viser vel i network tab viss ein ønsker å sjå det

Foreslår heller noko slikt: 

`return await this.client-request....`

Reply:
Godt poeng. Ser på det 👍